lib/active_waiter/enumerable_job.rb

```diff
@@ -0,0 +1,24 @@
+module ActiveWaiter::EnumerableJob
+  extend ActiveSupport::Concern
+
+  included do
+    include ActiveWaiter::Job
+  end
+
+  def perform(*args)
+    before(*args)
+    enumerable.each_with_index do |item, index|
+      foreach(item)
+      update_active_waiter(percentage: (100 * (index+1.to_f) / items_count))
+    end
+    after
```

Comment:
`*args` was given to `before(*args)` earlier as an entry point for `perform(*args)`. if they are needed elsewhere (e.g. `enumerable`, `foreach`, `result`) implementers can store and manage inside `before`